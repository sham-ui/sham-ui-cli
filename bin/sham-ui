#!/usr/bin/env node

require( 'commander' )
    .version( require( '../package' ).version )
    .usage( '<command> [options]' )
    .command( 'new', 'generate a new project from a template' ).alias( 'n' )
    .command( 'upgrade', 'upgrade existed project from a template' ).alias( 'u' )
    .command( 'generate', 'generate a new component' ).alias( 'g' )
    .command( 'collect', 'collect filters & directives' ).alias( 'c' )
    .command( 'destroy', 'destroy component' ).alias( 'd' )
    .parse( process.argv );