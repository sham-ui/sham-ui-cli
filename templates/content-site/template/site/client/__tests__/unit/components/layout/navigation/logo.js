import { createDI } from 'sham-ui';
import hrefto from 'sham-ui-router/lib/href-to';
import LayoutNavigationLogo  from '../../../../../src/components/layout/navigation/logo.sfc';
import renderer from 'sham-ui-test-helpers';

it( 'renders correctly', () => {
    const DI = createDI();
    DI.bind( 'router', {
        generate: () => '/'
    } );

    const meta = renderer( LayoutNavigationLogo, {
        DI,
        directives: {
            hrefto
        }
    } );
    expect( meta.toJSON() ).toMatchSnapshot();
} );
