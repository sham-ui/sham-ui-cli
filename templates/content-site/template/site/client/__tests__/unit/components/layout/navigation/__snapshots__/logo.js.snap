// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "Options": Object {
    "appState": Object {
      "darkThemeEnabled": false,
      "routerResolved": false,
    },
  },
  "html": "
<div class=\\"logo\\">
  <a href=\\"/\\">
    <img src=\\"/images/logo-dark.png\\" alt=\\"logo\\">
  </a>
</div>
",
}
`;
